chain-impl-mockchain/src/ledger/evm.rs

```diff
@@ -99,9 +97,23 @@ impl Ledger {
                 // update ledger state
                 self.accounts = new_state.clone();
                 self.logs = new_logs.clone();
-                Ok(())
             }
         }
+
+        self.update_environment();
```

Comment:
Updating environment should be for every new block applying, not for every evm transaction.